### PR TITLE
fix: backport emitWarning helper

### PR DESCRIPTION
**Pull request by nbbeeken**
Unify the way we print out messages relating to upcoming changes
or potential misuse of the driver. Using the emitWarning API allows
users to configure when and how messages are shown.

(code from #2740)

NODE-2317, NODE-3114

### CODE REVIEW

File: lib/core/connection/msg.js [hunk 1]
```diff
@@ -196,7 +197,7 @@ class BinMsg {
     while (this.index < this.data.length) {
       const payloadType = this.data.readUInt8(this.index++);
       if (payloadType === 1) {
-        console.error('TYPE 1');
+        emitWarning('TYPE 1'); // ???
```
Comment (emadum) [inline lib/core/connection/msg.js:200]:
```suggestion
   // It was decided that no driver makes use of payload type 1
   throw new MongoError('OP_MSG Payload Type 1 detected unsupported protocol');
```

**Review by emadum — APPROVED** [review]
Added a suggestion about throwing a warning on type 1 op_msg payloads, like in 4.0. Otherwise LGTM!

**Review by durran — APPROVED** [review]
LGTM

**Comment by wayahead** [comment]
#2744 is serious impact to many projects, so when the new 3.6 version will be released to npm registry? Thanks.

**Comment by nbbeeken** [comment]
We hope to have a release out tomorrow! (March 16)

**Comment by wayahead** [comment]
@nbbeeken appreciate your guys effort!

**Pull request merged**